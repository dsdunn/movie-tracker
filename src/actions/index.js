export const addMovies = (movies) => ({
  type: "ADD_MOVIES",
  movies
});

export const toggleFav = (id) => {
  type: "TOGGLE_FAV",
  id
};

export const toggleFilter = () => {
  type: "TOGGLE_FILTER"
};