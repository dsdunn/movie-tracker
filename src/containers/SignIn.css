.sign-in-form {
  position: absolute;
  display: flex;
  flex-direction: column;
  top: .2em;
  margin: 0 1em;
}

.sign-in-form input {
  padding: .3em 1em;
  font-size: .7em;
  margin: .2em 0;
  border-radius: 8px;
  border: 0;
  font-size: .9em;
}

.sign-in-form button {
  width: 20em;
  border: 0;
  border-radius: 8px;
  margin: .2em 0;
  font-size: 1em;
}