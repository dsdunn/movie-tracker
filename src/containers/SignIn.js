import React, { Component } from 'react';
import {Link, withRouter, Redirect, Route} from 'react-router-dom';
import { loginUser, logOutUser } from '../actions';
import { connect } from 'react-redux';
import { logIn } from '../api-calls';

export class SignIn extends Component {
  constructor(props) {
    super(props)
    this.state={
      email: '',
      password: '',
      hasError: false
    }
  }

  handleChange = (input) => {
    const { type, value } = input.target
    this.setState({
      [type]: value
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const user = await logIn(this.state);
    
    if (user) {
      this.props.login(user);
      this.setState({ hasError: false })    
    } else {
      this.setState({ hasError: true })
    }
  }

  isLoggedIn = () => {
    return this.state.hasError ? 
    <h5>email and password do not match</h5> :
    <div></div>
  }

  logOutUser = () => {
    this.props.logOutUser()
  }

  toLogOut = () => {
    console.log(this.props)
    return this.props.user.name ?
    <button onClick={this.logOutUser}>Log Out</button> :
    <div></div>
  }

  render() {
    return(
      <form onSubmit={this.handleSubmit}>
        <label htmlFor='email'>E-Mail</label>
        <input id= 'email' type='email' onChange={this.handleChange}/>
        <label htmlFor='password'>Password</label>
        <input id='password' type='password' onChange={this.handleChange}/>
        <button>Sign In</button>
        {this.isLoggedIn()}
        {this.toLogOut()}
      </form>
    );
  }
}

const mapStateToProps = (state) => ({
  user: state.user
})
const mapDispatchToProps = (dispatch) => ({
  login: (user) => dispatch(loginUser(user)),
  logOutUser: () => dispatch(logOutUser())
})

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
