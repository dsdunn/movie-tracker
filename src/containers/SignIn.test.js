import React from 'react';
import { SignIn, mapStateToProps, mapDispatchToProps } from '../containers/SignIn.js';
import * as action from '../actions';
import {shallow} from 'enzyme';


describe ('SignIn', () => {
  let wrapper
  let mockPopulateUserFavs
  let mocklogOutUser
  let mockLogin

  beforeEach(() => {
    mockLogin = jest.fn();
    mockPopulateUserFavs = jest.fn();
    mocklogOutUser = jest.fn();
    wrapper = shallow(<SignIn 
      login={mockLogin} 
      populateUserFavs={mockPopulateUserFavs} 
      logOutUser={mocklogOutUser} 
      user={{name: 'Joel'}}/>)

  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('has a default state of email, password, and hasError', () => {
 
    expect(wrapper.state('email')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
    expect(wrapper.state('hasError')).toEqual(false);
  })

  it('changes state.email onChange', () => {
    const mockEvent = {target: {value: 'bestTime', id:'email'}}
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('email')).toEqual(mockEvent.target.value)
  })

  it('changes state.password onChange', () => {
    const mockEvent = {target: {value: 'dumpsterFire', id: 'password'}}
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('password')).toEqual(mockEvent.target.value)
  })

})

