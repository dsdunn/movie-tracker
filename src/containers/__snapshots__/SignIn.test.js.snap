// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    logOutUser={[Function]}
    login={[Function]}
    populateUserFavs={[Function]}
    user={
        Object {
            "name": "Joel",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id="email"
          onChange={[Function]}
          placeholder="E-Mail"
          type="email"
/>,
        <input
          id="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
/>,
        <button>
          Sign In
</button>,
        <Link
          component={[Function]}
          replace={false}
          to="/signup"
>
          <button>
                    Sign Up
          </button>
</Link>,
        <div />,
        <Redirect
          push={false}
          to="/"
/>,
      ],
      "className": "sign-in-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "email",
          "onChange": [Function],
          "placeholder": "E-Mail",
          "type": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Sign Up
</button>,
          "component": [Function],
          "replace": false,
          "to": "/signup",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="email"
            onChange={[Function]}
            placeholder="E-Mail"
            type="email"
/>,
          <input
            id="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
/>,
          <button>
            Sign In
</button>,
          <Link
            component={[Function]}
            replace={false}
            to="/signup"
>
            <button>
                        Sign Up
            </button>
</Link>,
          <div />,
          <Redirect
            push={false}
            to="/"
/>,
        ],
        "className": "sign-in-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "onChange": [Function],
            "placeholder": "E-Mail",
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Sign Up
</button>,
            "component": [Function],
            "replace": false,
            "to": "/signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
